src/modules/usv_omni_control/usv_omni_control.cpp

```diff
@@ -139,15 +144,15 @@ void USVOmniControl::handleManualInputs(const manual_control_setpoint_s &manual_
 
 		} else {
 			/* direct control */
-		_thrust_setpoint.setAll(0.0f);
-		_thrust_setpoint(0) = manual_control_setpoint.throttle;
-		_thrust_setpoint(1) = manual_control_setpoint.roll;
+			_thrust_setpoint.setAll(0.0f);
+			_thrust_setpoint(0) = normalizeJoystickThrottleInput(manual_control_setpoint.throttle);
```

Comment:
For the record, I have been bitten by an issue that was solved here https://github.com/PX4/PX4-Autopilot/pull/20885
This wasn't present in uuv_att_control. I have added this as a separate function. This approach could be replicated in other control modules, as it at a glance explains what is happening with the throttle setpoint. This is my way of fighting with such complex functions: smaller functions.